pandemic/model.py

```diff
@@ -393,430 +400,234 @@ def pandemic_multiple_time_steps(
+            date_list=date_list,
+        )
+
+        # If time steps are monthly, aggregate predictions to
+        # annual for dashboard display
+        if len(date_list[i]) > 4:
+            print("aggregating monthly predictions to annual time steps...")
+            aggregate_monthly_output_to_annual(
+                formatted_geojson=full_out_df, outpath=outpath
+            )
+
+        # Save model metadata to text file
+        print("writing model metadata...")
+        main_model_output = e[0]
+        final_presence_col = sorted(
+            [c for c in main_model_output.columns if c.startswith("Presence")]
         )[-1]
+        meta = {}
+        meta["PARAMETERS"] = []
+        meta["PARAMETERS"].append(
+            {
+                "alpha": str(alpha),
+                "beta": str(beta),
+                "mu": str(mu),
+                "lamda_c": str(lamda_c),
+                "phi": str(phi),
+                "sigma_epsilon": str(sigma_epsilon),
+                "sigma_h": str(sigma_h),
+                "sigma_kappa": str(sigma_kappa),
+                "sigma_phi": str(sigma_phi),
+                "sigma_T": str(sigma_T),
+                "start_year": str(start_year),
+                "random_seed": str(random_seed),
+            }
+        )
+        meta["NATIVE_COUNTRIES_T0"] = native_countries_list
+        meta["COMMODITY"] = commodities_available[i]
+        meta["FORECASTED"] = commodity_forecast_path
+        meta["PHYTOSANITARY_CAPACITY_WEIGHTS"] = phyto_dict
+        meta["TOTAL COUNTRIES INTRODUCTED"] = str(
+            main_model_output[final_presence_col].value_counts()[1]
+            - len(native_countries_list)
+        )
 
-    with open(f'{outpath}run{run_num}_meta.txt', 'w') as file:
-        file.write(f'PARAMETER VALS:\talpha: {alpha}' \
-                   f'\n\tbeta: {beta}\n\tmu: {mu}' \
-                   f'\tsigma_h: {sigma_h}\n\tsigma_kappa: {sigma_kappa}\n\t' \
-                   f'sigma_T: {sigma_T}\n\n')
-        file.write(f'NATIVE COUNTRIES AT T0:\n\t{native_countries_list}\n\n')
-        file.write(f'COMMODITIES: {commodity_path}')
-        file.write(f'\tForecasted: {commodity_forecast_path}')
-        file.write(f'PHYTOSANITARY CAPACITY:\n\t {gdp_path}')
-        file.write(f'\tGPD vals:{gdp_dict}\n\n')
-        file.write('COUNTRY INTRODUCTIONS:')
-        file.write(f'\nTotal Number of Countries: ' \
-                   f'{main_model_output[final_presence_col].value_counts()[1]}')
-        file.close()
-        print(f'saving: {outpath}run{run_num}_meta.txt')
-
-generate_model_metadata(
-    outpath = out_dir + f'/run{run_num}',
-    run_num = run_num,
-    native_countries_list = native_countries_list,
-    gdp_dict = gdp_dict, 
-    main_model_output = e[0]
-    )
+        with open(f"{outpath}/run{run_num}_meta.txt", "w") as file:
+            json.dump(meta, file, indent=4)
+
+    else:
+        print("\tskipping as pest is not transported with this commodity")
```

Comment:
Will lambda always be 0 or 1 or is it possible to be something in between, more like a commodity weight?

Reply:
Another good catch - will change that to > 0 instead of ==1